### PR TITLE
Updating GeoMasterDataProvider to work with slots

### PR DESCRIPTION
**Pull request by puneetg1983**

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 1]
```diff
@@ -67,9 +69,9 @@ private IGeoMasterClient InitClient()
         /// }
         /// </code>
         /// </example>
-        public async Task<IDictionary<string,string>> GetAppSettings(string subscriptionId, string resourceGroupName, string name)
+        public async Task<IDictionary<string,string>> GetAppSettings(string subscriptionId, string resourceGroupName, string name, string slotName)
```
Comment (hforeste) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: >string slotName [](start = 131, length = 15)

You can make this an optional parameter by setting this equal to a default value such as GeoMasterConstants.ProductionSlot. By doing that you don't have to create another GetAppSettings method.
And the same for the other new methods you added.
What you did is fine though, this is just a preference and something I typically see that C# developers do.

Comment (ShekharGupta1988) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Hawk - this is the correct thing to do in order to maintain backward compatibility. If you change the signature of the existing method (even with optional param), then detectors which are using tht method would need to recompiled.

Comment (hforeste) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Oh i see. Kool. Thanks for that clarification man 👍

Comment (puneetg1983) [reply src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Exactly what @ShekharGupta1988 said :-) 

I had it initially with Optional parameters but I found that existing DLLs were breaking so did it this way. 

Thanks for the review!

**Pull request merged**